### PR TITLE
Add time_window in the new experimental module

### PR DESCRIPTION
**Pull request by jmao-denver**
1. added experimental.py and time_window() in it
2. add test_experiment.py
3. make sure addTimeWindowInternal checks for upg ex lock at the very beginnning
4. removed the unused _Jcompat.py that's left from the py projects reorg work

### CODE REVIEW

File: py/server/deephaven/experimental.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+from deephaven.table import Table
+
+_JWindowCheck = jpy.get_type("io.deephaven.engine.util.WindowCheck")
+
+
+def time_window(table: Table, ts_col: str, window: int, bool_col: str) -> Table:
+    """Creates a new table by applying a time window to the source table and adding a new Boolean column.
+
+    The value of the new Boolean column is set to false when the timestamp column value is older than the window from
+    now or true otherwise. If the timestamp column value is null; the Boolean column value will be null as well. The
```
Comment (jcferretti) [inline py/server/deephaven/experimental.py:17]: Replace semicolon by comma

File: py/server/deephaven/experimental.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    resultant table ticks whenever the source table ticks, or modifies a row when it passes out of the window.
```
Comment (jcferretti) [inline py/server/deephaven/experimental.py:18]: Resultant sounds funny in my ears; resulting sounds better

**Review by jcferretti — APPROVED** [review]

**Review by hythloda — APPROVED** [review]

File: py/server/deephaven/_jcompat.py [hunk 3]
```diff
@@ -1,95 +0,0 @@
-#
-#   Copyright (c) 2016-2022 Deephaven Data Labs and Patent Pending
-#
-""" This module provides Java compatibility support including convenience functions to create some widely used Java
```
Comment (rcaudy) [inline py/server/deephaven/_jcompat.py:4]: I assume this was unused?

**Review by rcaudy — APPROVED** [review]

File: py/server/deephaven/_jcompat.py (hunk 3 above)
Comment (jmao-denver) [reply py/server/deephaven/_jcompat.py:4]: yes, we decided to make it public but the renaming change and the py projects reorg work somehow left both in

**Pull request merged**